vegeta_wrapper/trigger_vegeta.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import os
+import subprocess
+import json
+import dateutil.parser
+import logging
+
+logger = logging.getLogger("snafu")
+
+
+class Trigger_vegeta():
+    def __init__(self, args):
+        self.uuid = args.uuid
+        self.user = args.user
+        self.sample = args.sample
+        self.workers = args.workers
+        self.targets = args.targets
+        self.duration = args.duration
+        self.cluster_name = args.cluster_name
+        self.keepalive = args.keepalive
+
+    def _json_payload(self, data, sample):
+        payload = {
+            "workload": "vegeta",
+            "uuid": self.uuid,
+            "user": self.user,
+            "cluster_name": self.cluster_name,
+            "iteration": sample,
+            "duration": self.duration,
+            "workers": self.workers,
+            "keepalive": self.keepalive,
+        }
+        payload.update(data)
+        return payload
+
+    def _run_vegeta(self):
+        cmd = (
+            "vegeta attack -keepalive={0} -insecure -workers={1} -duration={2}s -targets={3} -rate=0"
```

Comment:
this will default to http2, should we allow that to be disabled?

Reply:
thinking more about this, i don't think it is necessary...

Reply:
According to the vegeta docs:

```
-http2
    	Send HTTP/2 requests when supported by the server (default true)
```

Considering this, vegeta will try to negotiate http2 and fallback to http/1.1. This is the normal behavior of almost every http client (web browsers, curl, etc).

Reply:
yup, hence me resolving the convo... too early to be doing reviews :P